extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientImpl.java

```diff
@@ -109,7 +109,7 @@ public Uni<Tokens> get() {
                         body.add(OidcConstants.CLIENT_ASSERTION, jwt);
                     }
                 } else if (!OidcCommonUtils.isClientSecretPostAuthRequired(oidcConfig.credentials)) {
-                    body.add(OidcConstants.CLIENT_ID, oidcConfig.clientId.get());
+                    body = copyMultiMap(body).add(OidcConstants.CLIENT_ID, oidcConfig.clientId.get());
```

Comment:
Isn't changing to `set` enough here?

Reply:
@gastaldi Sorry, forgot to add a comment relating to `set`, since `OidcClient` will be reused by multiple threads, I decided to avoid `set`

Reply:
Copying the MultiMap is fine, my question was more that if you call `add` again you will end up with more than one ClientID in the Map?

Reply:
Hey @gastaldi, sorry for asking to review on Sunday :-).
I see,  so, in the original map client id is not present,  this copy is created per every request and I guess, either `set` or `add` will do. I've added a test, as you can see in the original issue the duplication starts from a 2nd request (double `add`), so the test does 2 requests, without the fix the 2nd request was failing.
if you'd like I can replace `add` with `set`, would not be a problem at all

Reply:
No biggie, just curious really ;)

Reply:
Hi George, just pushed a minor update, it is quiet today, so an extra CI run will not be a problem, `set` reads better for setting a single value property :-), thanks for the time